features/csharp/csharp_persistence.feature

```diff
@@ -6,7 +6,9 @@ Feature: Unity Persistence
   @skip_windows @skip_macos @skip_webgl #pending PLAT-8632
   Scenario: Receive a persisted session
     When I run the game in the "PersistSession" state
+    And I set the HTTP status code for the next requests to "408"
     And I wait for requests to fail
+    And I discard the oldest session
```

Comment:
I would add a `I wait to receive an error` before discarding the oldest session, but after `I wait for requests to fail`.

It would just add some confidence that the notifier really did try to make a request - and avoid a potentially confusing/messy failure when trying to discard something from an empty list (if one isn't sent).

You might find the `I wait for requests to fail` is no longer needed (and would be great to remove), but as discussed IRL a ticket to look at that separately would be fine.

Reply:
I decided to do this properly as part of this PR.
I have done everything suggested and i changed `I wait for requests to fail` to `I wait for requests to persist` which is just a 1 second sleep, i think that should be enough.